src/deepEqual.js

```diff
@@ -23,8 +23,48 @@
  *
  */
 
+function isEquals(value1, value2) {
+  return typeof value1 === typeof value2 && value1 === value2;
+}
+
+function isObject(elem, elem2) {
+  let result;
+  if (elem === null || elem2 === null) {
+    result = false;
+  } else if (typeof elem === 'object') {
+    result = true;
+  } else {
+    result = false;
+  }
+  return result;
+}
+
+function objLength(obj1, obj2) {
+  return Object.keys(obj1).length === Object.keys(obj2).length;
+}
+
 function deepEqual(a, b) {
-  // write code here
+  if (isObject(a, b)) {
+    if (objLength(a, b)) {
+      for (let key in a) {
+        let keyOfA = a[key];
+        let keyOfB = b[key];
+        if (isObject(keyOfA, keyOfB) && deepEqual(keyOfA, keyOfB)) {
+          continue;
+        }
+        if (Object.keys(b).includes(key)) {
+          if (keyOfA !== keyOfB) {
+            return false;
+          }
+          continue;
+        }
+        return false;
```

Comment:
`if (Object.keys(b).includes(key)) {
          if (keyOfA !== keyOfB) {
            return false;
          }
          continue;
        }
        return false;`

this should be equivalent to 
if (keyOfA !== keyOfB) {
  return false;
}